libbeat/template/load.go

```diff
@@ -31,131 +31,185 @@ import (
 	path := "/_template/" + templateName
-	body, err := loadJSON(l.client, path, template)
+	resp, err := loadJSON(l.client, path, template)
 	if err != nil {
-		return fmt.Errorf("couldn't load template: %v. Response body: %s", err, body)
+		return fmt.Errorf("couldn't load template: %v. Response body: %s", err, resp)
 	}
-	logp.Info("Elasticsearch template with name '%s' loaded", templateName)
 	return nil
 }
 
-// CheckTemplate checks if a given template already exist. It returns true if
+// exists checks if a given template already exist. It returns true if
 // and only if Elasticsearch returns with HTTP status code 200.
-func (l *Loader) CheckTemplate(templateName string) bool {
+func (l *esLoader) templateExists(templateName string) bool {
+	if l.client == nil {
+		return false
+	}
 	status, _, _ := l.client.Request("HEAD", "/_template/"+templateName, "", nil, nil)
-	return status == 200
+
+	if status != 200 {
+		return false
+	}
+
+	return true
+}
+
+type stdoutLoader struct {
+	info      beat.Info
+	fields    []byte
+	migration bool
+}
+
+func (l *stdoutLoader) Load(config TemplateConfig) error {
+	//build template from config
+	tmpl, err := template(config, l.info, common.Version{}, l.migration)
+	if err != nil || tmpl == nil {
+		return err
+	}
+
+	//create body to print
+	body, err := buildBody(l.info, tmpl, config, l.fields)
+	if err != nil {
+		return err
+	}
+
+	p := common.MapStr{tmpl.name: body}
+	str := fmt.Sprintf("%s\n", p.StringToPrint())
+	if _, err := os.Stdout.WriteString(str); err != nil {
+		return fmt.Errorf("error printing template: %v", err)
+	}
+	return nil
+}
+
+func template(config TemplateConfig, info beat.Info, esVersion common.Version, migration bool) (*Template, error) {
+	if !config.Enabled {
+		logp.Info("template config not enabled")
+		return nil, nil
+	}
+	tmpl, err := New(info.Version, info.IndexPrefix, esVersion, config, migration)
+	if err != nil {
+		return nil, fmt.Errorf("error creating template instance: %v", err)
+	}
+	return tmpl, nil
+}
+
+func buildBody(info beat.Info, tmpl *Template, config TemplateConfig, fields []byte) (common.MapStr, error) {
+	if config.Overwrite {
+		logp.Info("Existing template will be overwritten, as overwrite is enabled.")
+	}
+
+	var err error
+	var body map[string]interface{}
+	if config.JSON.Enabled {
+		jsonPath := paths.Resolve(paths.Config, config.JSON.Path)
+		if _, err = os.Stat(jsonPath); err != nil {
+			return nil, fmt.Errorf("error checking json file %s for template: %v", jsonPath, err)
+		}
+		logp.Info("Loading json template from file %s", jsonPath)
+		content, err := ioutil.ReadFile(jsonPath)
+		if err != nil {
+			return nil, fmt.Errorf("error reading file %s for template: %v", jsonPath, err)
+
+		}
+		err = json.Unmarshal(content, &body)
+		if err != nil {
+			return nil, fmt.Errorf("could not unmarshal json template: %s", err)
+		}
+
+		// Load fields from path
+	} else if config.Fields != "" {
```

Comment:
Better move the comment `// Load fields from path` behind the if statment or into the current block. Consider splitting this into separate functions. But the place of the comment in the other block is quite confusing.